hoot-core/src/main/cpp/hoot/core/criterion/PolygonCriterion.cpp

```diff
@@ -60,24 +62,14 @@ bool PolygonCriterion::isSatisfied(const ConstElementPtr& e) const
   else if (e->getElementType() == ElementType::Way)
   {
     ConstWayPtr way = std::dynamic_pointer_cast<const Way>(e);
-    if (!way->isClosedArea())
-    {
-      return false;
-    }
-  }
-
-  const Tags& t = e->getTags();
-  for (Tags::const_iterator it = t.constBegin(); it != t.constEnd(); ++it)
-  {
-    const SchemaVertex& tv = OsmSchema::getInstance().getTagVertex(it.key() + "=" + it.value());
-    uint16_t g = tv.geometries;
-    if (g & (OsmGeometries::ClosedWay | OsmGeometries::Area))
+    if (way->isValidPolygon() && way->isClosedArea())
```

Comment:
Simplified this....seems to work fine. The extra area schema checks are only applicable if you expect the element to be identified as area in the schema. Since this identifies generic polys, the checks aren't needed.